ksqldb-parser/src/test/java/io/confluent/ksql/parser/tree/CreateConnectorTest.java

```diff
@@ -39,18 +39,18 @@
   public void testEquals() {
     new EqualsTester()
         .addEqualityGroup(
-            new CreateConnector(Optional.of(SOME_LOCATION), NAME, CONFIG, CreateConnector.Type.SOURCE),
-            new CreateConnector(Optional.of(OTHER_LOCATION), NAME, CONFIG, CreateConnector.Type.SOURCE),
-            new CreateConnector(NAME, CONFIG, CreateConnector.Type.SOURCE)
+            new CreateConnector(Optional.of(SOME_LOCATION), NAME, CONFIG, CreateConnector.Type.SOURCE,false),
+            new CreateConnector(Optional.of(OTHER_LOCATION), NAME, CONFIG, CreateConnector.Type.SOURCE,false),
+            new CreateConnector(NAME, CONFIG, CreateConnector.Type.SOURCE,false)
         )
         .addEqualityGroup(
-            new CreateConnector(OTHER_NAME, CONFIG, CreateConnector.Type.SOURCE)
+            new CreateConnector(OTHER_NAME, CONFIG, CreateConnector.Type.SOURCE,false)
         )
         .addEqualityGroup(
-            new CreateConnector(NAME, OTHER_CONFIG, CreateConnector.Type.SOURCE)
+            new CreateConnector(NAME, OTHER_CONFIG, CreateConnector.Type.SOURCE,false)
         )
         .addEqualityGroup(
-            new CreateConnector(NAME, CONFIG, CreateConnector.Type.SINK)
+            new CreateConnector(NAME, CONFIG, CreateConnector.Type.SINK,false)
```

Comment:
Please add test case on the end:

```java
.addEqualityGroup(
   new CreateConnector(NAME, CONFIG, CreateConnector.Type.SINK, true)
)
```

This would have found the bug of not having this new field in `hashCode` and `equals` :D

Reply:
Indeed yes , :) I was under the impression object equality here should not depend on `ifNotExists`, 
but I see the rationale now .. added the test case